### PR TITLE
Improved the consistency of the Grunt build with the themes

### PR DESCRIPTION
**Pull request by LaurentGoderre**
Related PRs

* wet-boew/theme-base#136
* wet-boew/theme-gc-intranet#166
* wet-boew/theme-gcwu-fegc#246
* wet-boew/theme-ogpl#127
* wet-boew/GCWeb#1231

### CODE REVIEW

**Comment by LaurentGoderre** [comment]
@nschonni are you ok with this?

File: Gruntfile.coffee [hunk 1]
```diff
@@ -334,6 +334,14 @@ module.exports = (grunt) ->
 
 		deployBranch: "v4.0-dist"
 
+		checkDependencies:
+			all:
+				options:
+					npmInstall: false
+
+		clean:
```
Comment (nschonni) [inline Gruntfile.coffee]: Any reason to move this?

Comment (nschonni) [inline Gruntfile.coffee]: Any reason to move this?

File: Gruntfile.coffee [hunk 2]
```diff
@@ -334,6 +334,14 @@ module.exports = (grunt) ->
 
 		deployBranch: "v4.0-dist"
 
+		checkDependencies:
```
Comment (nschonni) [inline Gruntfile.coffee]: Any reason to move this?

File: Gruntfile.coffee [hunk 3]
```diff
@@ -301,10 +301,10 @@ module.exports = (grunt) ->
 
 	grunt.util.linefeed = "\n"
 	# Project configuration.
-	grunt.initConfig
+	@initConfig
 
 		# Metadata.
-		pkg: grunt.file.readJSON("package.json")
+		pkg: @file.readJSON("package.json")
```
Comment (nschonni) [inline Gruntfile.coffee]: If you're going to change `grunt` to `@`, it should probably be done everywhere. 3 lines below is the same thing, but not changed

**Review by nschonni — CHANGES_REQUESTED** [review]
Some nits and quetions

File: Gruntfile.coffee (hunk 3 above)
Comment (LaurentGoderre) [inline Gruntfile.coffee]: All of these were done to minimize the diff between this repo and the themes.

Comment (nschonni) [inline Gruntfile.coffee]: That's fine, then it should be done consistently and pushed down to the themes too

**Comment by LaurentGoderre** [comment]
@nschonni Can you review again?

File: Gruntfile.coffee [hunk 4]
```diff
@@ -403,18 +411,18 @@ module.exports = (grunt) ->
 				options:
 					banner: ""
 					process: ( src, filepath ) ->
-						lang = filepath.replace grunt.config( "coreDist" ) + "/js/i18n/", ""
+						lang = filepath.replace( grunt.config "coreDist" + "/js/i18n/", "")
```
Comment (nschonni) [inline Gruntfile.coffee]: This seems like it is actually changing behaviour by dropping the braces.

File: Gruntfile.coffee [hunk 5]
```diff
@@ -403,18 +411,18 @@ module.exports = (grunt) ->
...
 						# jQuery validation uses an underscore for locals
 						lang = lang.replace "_", "-"
 						validationPath = "lib/jquery-validation/src/localization/"
 
 						# Check and append message file
 						messagesPath = validationPath + "messages_" + lang
-						messages = if grunt.file.exists messagesPath then grunt.file.read( messagesPath ) else ""
+						messages = if grunt.file.exists messagesPath then grunt.file.read messagesPath else ""
```
Comment (nschonni) [inline Gruntfile.coffee]: More `grunt.` stuff that I guess should be `@.` if that's the pattern you're going to go with

**Review by nschonni — DISMISSED** [review]
Not sure dropping the braces actually improves readablitiy

File: Gruntfile.coffee (hunk 5 above)
Comment (LaurentGoderre) [inline Gruntfile.coffee]: No, I tried converting those and it wouldn't work

File: Gruntfile.coffee (hunk 4 above)
Comment (LaurentGoderre) [inline Gruntfile.coffee]: Does it?

Comment (LaurentGoderre) [inline Gruntfile.coffee]: You're right

**Review by lucas-hay — APPROVED** [review]
Laurent has fixed as per Nick's requests.

**Review by lucas-hay — APPROVED** [review]

**Pull request merged**